--- src/lib/ViZDoomController.cpp
+++ src/lib/ViZDoomController.cpp
@@ -238,7 +238,7 @@
         try{
             if (this->doomRunning) {
                 // Try to stop demo recording
-                if (this->gameState->DEMO_RECORDING){
+                if (this->gameState && this->gameState->DEMO_RECORDING){
                     this->sendCommand("stop");
                     this->MQDoom->send(MSG_CODE_TIC);
                     this->waitForDoomWork();
