--- python/private/py_repositories.bzl	2026-03-21 21:54:45
+++ python/private/py_repositories.bzl	2026-03-21 21:54:45
@@ -55,16 +55,16 @@
     )
     http_archive(
         name = "rules_cc",
-        sha256 = "d9bdd3ec66b6871456ec9c965809f43a0901e692d754885e89293807762d3d80",
-        strip_prefix = "rules_cc-0.0.13",
-        urls = ["https://github.com/bazelbuild/rules_cc/releases/download/0.0.13/rules_cc-0.0.13.tar.gz"],
+        sha256 = "bbf1ae2f83305b7053b11e4467d317a7ba3517a12cef608543c1b1c5bf48a4df",
+        strip_prefix = "rules_cc-0.0.16",
+        urls = ["https://github.com/bazelbuild/rules_cc/releases/download/0.0.16/rules_cc-0.0.16.tar.gz"],
     )
 
     # Needed by rules_cc, triggerred by @rules_java_prebuilt in Bazel by using @rules_cc//cc:defs.bzl
     http_archive(
         name = "protobuf",
-        sha256 = "da288bf1daa6c04d03a9051781caa52aceb9163586bff9aa6cfb12f69b9395aa",
-        strip_prefix = "protobuf-27.0",
-        url = "https://github.com/protocolbuffers/protobuf/releases/download/v27.0/protobuf-27.0.tar.gz",
+        sha256 = "e4e6ff10760cf747a2decd1867741f561b216bd60cc4038c87564713a6da1848",
+        strip_prefix = "protobuf-34.1",
+        url = "https://github.com/protocolbuffers/protobuf/releases/download/v34.1/protobuf-34.1.tar.gz",
     )
     pypi_deps()
