--- python/private/py_executable.bzl
+++ python/private/py_executable.bzl
@@ -1014,9 +1014,10 @@
     manifest.set_param_file_format("multiline")
 
     manifest.add("__main__.py={}".format(zip_main.path))
     manifest.add("__init__.py=")
-    manifest.add(
-        "{}=".format(_get_zip_runfiles_path(_INIT_PY, workspace_name)),
-    )
-
+    if _should_create_init_files(ctx):
+        manifest.add(
+            "{}=".format(_get_zip_runfiles_path(_INIT_PY, workspace_name)),
+        )
+
     def map_zip_empty_filenames(list_paths_cb):
