--- tools/re2c/substr.cc
+++ tools/re2c/substr.cc
@@ -4,36 +4,32 @@
 #include "substr.h"
 #include "globals.h"

-#ifndef HAVE_STRNDUP
-
-char *strndup(const char *str, size_t len)
+static char *re2c_strndup(const char *str, size_t len)
 {
  char * ret = (char*)malloc(len + 1);
+	if (ret == NULL)
+		return NULL;

  memcpy(ret, str, len);
  ret[len] = '\0';
  return ret;
 }

-#endif
-
 namespace re2c
 {

 void SubStr::out(std::ostream& o) const
 {
  o.write(str, len);
 }

 bool operator==(const SubStr &s1, const SubStr &s2)
 {
  return (bool) (s1.len == s2.len && memcmp(s1.str, s2.str, s1.len) == 0);
 }

 Str::Str(const SubStr& s)
-	//VIZDOOM_CODE
-	//: SubStr(strndup(s.str, s.len), s.len)
-	: SubStr(strdup(s.str), s.len)
+	: SubStr(re2c_strndup(s.str, s.len), s.len)
 {
  ;
 }
