--- src/win32/i_system.h
+++ src/win32/i_system.h
@@ -23,6 +23,10 @@
 #ifndef __I_SYSTEM__
 #define __I_SYSTEM__
 
+#define USE_WINDOWS_DWORD
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
 #include "doomtype.h"
 
 struct ticcmd_t;
--- src/win32/i_system.cpp
+++ src/win32/i_system.cpp
@@ -1309,7 +1309,7 @@
 	HDC dc = GetDC(NULL);
 	if (dc == NULL)
 	{
-		return false;
+		return NULL;
 	}
 	HDC and_mask_dc = CreateCompatibleDC(dc);
 	HDC xor_mask_dc = CreateCompatibleDC(dc);
