--- build_defs.bzl
+++ build_defs.bzl
@@ -6,8 +6,6 @@
 """Build rules for pybind11."""
 
 load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
-load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
-load("@rules_cc//cc:cc_library.bzl", "cc_library")
-load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 def _pybind_py_env_test_impl(ctx):
