--- build_defs.bzl
+++ build_defs.bzl
@@ -6,9 +6,7 @@
 """Build rules for pybind11."""
 
 load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
-load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
-load("@rules_cc//cc:cc_library.bzl", "cc_library")
-load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 def _pybind_py_env_test_impl(ctx):
     toolchain = ctx.toolchains["@rules_python//python:toolchain_type"]
@@ -158,7 +156,7 @@
     )
 
     cc_binary(
-        name = name + ".pyd",
+        name = name + "_dll",
         copts = copts + PYBIND_COPTS + select({
             Label("@pybind11//:msvc_compiler"): [],
             "//conditions:default": ["-fvisibility=hidden"],
@@ -177,6 +175,20 @@
         }),
         deps = deps + PYBIND_DEPS,
         **kwargs
+    )
+
+    copy_file(
+        name = name + ".pyd",
+        src = ":" + name + "_dll",
+        out = name + ".pyd",
+        allow_symlink = True,
+        tags = tags,
+        target_compatible_with = select({
+            "@platforms//os:windows": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+        testonly = kwargs.get("testonly"),
+        visibility = kwargs.get("visibility"),
     )
 
     native.alias(
