+++ game_factory.h
@@ -0,0 +1,20 @@
+#include "game.h"
+#ifndef GLOBAL_GAME_FACTORY_H
+#define GLOBAL_GAME_FACTORY_H
+std::shared_ptr<Game> make_bigfish(void);
+std::shared_ptr<Game> make_bossfight(void);
+std::shared_ptr<Game> make_caveflyer(void);
+std::shared_ptr<Game> make_chaser(void);
+std::shared_ptr<Game> make_climber(void);
+std::shared_ptr<Game> make_coinrun(void);
+std::shared_ptr<Game> make_dodgeball(void);
+std::shared_ptr<Game> make_fruitbot(void);
+std::shared_ptr<Game> make_heist(void);
+std::shared_ptr<Game> make_jumper(void);
+std::shared_ptr<Game> make_leaper(void);
+std::shared_ptr<Game> make_maze(void);
+std::shared_ptr<Game> make_miner(void);
+std::shared_ptr<Game> make_ninja(void);
+std::shared_ptr<Game> make_plunder(void);
+std::shared_ptr<Game> make_starpilot(void);
+#endif
