--- boost.BUILD
+++ boost.BUILD
@@ -852,6 +852,7 @@
         ":predef",
         ":range",
+        ":scope",
         ":scoped_array",
         ":smart_ptr",
         ":static_assert",
@@ -1758,6 +1759,15 @@
     ],
 )
 
+boost_library(
+    name = "scope",
+    deps = [
+        ":config",
+        ":core",
+        ":type_traits",
+    ],
+)
+
 boost_library(
     name = "scope_exit",
     deps = [
