--- lzma.BUILD.orig
+++ lzma.BUILD
@@ -69,25 +69,22 @@
     copts = select({
         "@platforms//os:windows": [],
         "//conditions:default": ["-std=c99"],
-    }) + [
-        # Replace with local_includes if it's ever added https://github.com/bazelbuild/bazel/issues/16472
-        "-I external/org_lzma_lzma/src/common",
-        "-I external/org_lzma_lzma/src/liblzma",
-        "-I external/org_lzma_lzma/src/liblzma/api",
-        "-I external/org_lzma_lzma/src/liblzma/common",
-        "-I external/org_lzma_lzma/src/liblzma/check",
-        "-I external/org_lzma_lzma/src/liblzma/delta",
-        "-I external/org_lzma_lzma/src/liblzma/lz",
-        "-I external/org_lzma_lzma/src/liblzma/lzma",
-        "-I external/org_lzma_lzma/src/liblzma/rangecoder",
-        "-I external/org_lzma_lzma/src/liblzma/simple",
-    ],
+    }),
     defines = select({
         "@platforms//os:windows": ["LZMA_API_STATIC"],
         "//conditions:default": [],
     }),
     includes = [
+        "src/common",
+        "src/liblzma",
         "src/liblzma/api",
+        "src/liblzma/common",
+        "src/liblzma/check",
+        "src/liblzma/delta",
+        "src/liblzma/lz",
+        "src/liblzma/lzma",
+        "src/liblzma/rangecoder",
+        "src/liblzma/simple",
     ],
     linkopts = select({
         "@platforms//os:android": [],
