diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,10 +93,6 @@ set(SDL2COMPAT_VENDOR_INFO "" CACHE STRING "Vendor name and/or version to add to
 set(SDL2COMPAT_FRAMEWORK FALSE)
 #FIXME: should SDL2-compat modify vendor_info/revision information?
 
-if(SDL2COMPAT_STATIC AND NOT (CMAKE_SYSTEM_NAME MATCHES "Linux"))
-  message(FATAL_ERROR "Static builds are only supported on Linux.")
-endif()
-
 macro(check_add_debug_flag FLAG SUFFIX)
   check_c_compiler_flag(${FLAG} HAS_C_FLAG_${SUFFIX})
   if(HAS_C_FLAG_${SUFFIX})
@@ -490,7 +486,7 @@ if(SDL2COMPAT_STATIC)
       "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
       "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>"
   )
-  target_compile_definitions(SDL2-static PRIVATE _REENTRANT)
+  target_compile_definitions(SDL2-static PRIVATE _REENTRANT SDL2COMPAT_STATIC_LINK)
   target_compile_definitions(SDL2-static PRIVATE "SDL2COMPAT_REVISION=\"${SDL2COMPAT_REVISION}\"")
   target_link_libraries(SDL2-static PRIVATE ${CMAKE_DL_LIBS})
   set_target_properties(SDL2-static PROPERTIES
diff --git a/src/dynapi/SDL_dynapi.h b/src/dynapi/SDL_dynapi.h
--- a/src/dynapi/SDL_dynapi.h
+++ b/src/dynapi/SDL_dynapi.h
@@ -74,7 +74,11 @@
 
 /* everyone else. This is where we turn on the API if nothing forced it off. */
 #ifndef SDL_DYNAMIC_API
+#ifdef SDL2COMPAT_STATIC_LINK
+#define SDL_DYNAMIC_API 0
+#else
 #define SDL_DYNAMIC_API 1
+#endif
 #endif
 
 #if SDL_DYNAMIC_API
diff --git a/src/sdl2_compat.c b/src/sdl2_compat.c
--- a/src/sdl2_compat.c
+++ b/src/sdl2_compat.c
@@ -118,6 +118,12 @@
 extern "C" {
 #endif
 
+#ifdef SDL2COMPAT_STATIC_LINK
+#define SDL3_SYM(rc,fn,params,args,ret) \
+    extern rc SDLCALL ENVPOOL_SDL3_SDL_##fn params;
+#include "sdl3_syms.h"
+#endif
+
 /** Define SDL2COMPAT_TEST_SYMS=1 to have warnings about wrong prototypes of src/sdl3_sym.h
  *  It won't compile but it helps to make sure it's sync'ed with SDL3 headers.
  */
@@ -365,7 +371,26 @@ static bool SDL2Compat_CheckDebugLogging(void)
 /* Obviously we can't use SDL_LoadObject() to load SDL3.  :)  */
 /* FIXME: Updated library names after https://github.com/libsdl-org/SDL/issues/5626 solidifies.  */
 static char loaderror[256];
-#if defined(_WIN32)
+#ifdef SDL2COMPAT_STATIC_LINK
+    #ifdef __linux__
+    #include <dlfcn.h>
+    #endif
+    static bool Loaded_SDL3 = false;
+    static void *LookupSDL3Sym(const char *symbol)
+    {
+        #define SDL3_SYM(rc,fn,params,args,ret) \
+            if (SDL2Compat_strequal(symbol, "SDL_" #fn)) { \
+                return (void *)ENVPOOL_SDL3_SDL_##fn; \
+            }
+        #include "sdl3_syms.h"
+        return NULL;
+    }
+    #define LoadSDL3Library() (Loaded_SDL3 = true)
+    #define CloseSDL3Library() { Loaded_SDL3 = false; }
+    #ifdef _WIN32
+    #define DIRSEP "\\"
+    #endif
+#elif defined(_WIN32)
     static HMODULE Loaded_SDL3 = NULL;
     #define DIRSEP "\\"
     #define SDL3_LIBNAME "SDL3.dll"
